uc-crux-llvm/src/UCCrux/LLVM/Config/FromEnv.hs

```diff
@@ -58,13 +68,34 @@ suggestedLoopBound = 8
 suggestedRecursionBound :: Word64
 suggestedRecursionBound = 8
 
+-- | Parse (as in \"parse, don\'t validate\") options gathered by Crux
+-- ('UCCruxLLVMOptions') into an 'AppContext' and a 'TopLevelConfig'.
 processUCCruxLLVMOptions ::
-  (CruxOptions, UCCruxLLVMOptions) -> IO (AppContext, CruxOptions, UCCruxLLVMOptions)
+  (CruxOptions, UCCruxLLVMOptions) -> IO (AppContext, CruxOptions, TopLevelConfig)
 processUCCruxLLVMOptions (initCOpts, initUCOpts) =
   do
     let appCtx = makeAppContext (verbosityFromInt (verbosity initUCOpts))
-    when (not (doExplore initUCOpts) && null (entryPoints initUCOpts)) $
-      die "At least one entry point (--entry-points) is required (or try --explore)"
+    let doCrashEquivalence = crashEquivalence initUCOpts /= ""
+
+    entries <-
+      if doExplore initUCOpts
+      then pure Nothing
+      else
+        if doCrashEquivalence
+        then pure (nonEmpty (entryPoints initUCOpts))
+        else
+          Just <$>
+            maybe
+              (die
+                (unwords
+                  [ "At least one entry point (--entry-points) is required",
+                    "(or try --explore or --crash-equivalence)"
+                  ]))
+              pure
+              (nonEmpty (entryPoints initUCOpts))
+
+    when (doExplore initUCOpts && doCrashEquivalence) $
+      die "Can't specify both --explore and --crash-equivalence"
```

Comment:
Is this function something that might be used in a library context? If so, `die` is probably too aggressive here (exiting without chance for recovery)

Reply:
I guess on closer inspection it doesn't call `exit(2)` - it does just throw an exception. If we are librarifying this, though, should we be using a more library-specific exception type to make it easier to integrate?

Reply:
I've moved this module under `Main` and added commentary saying that it's not really useful as part of a library API.